src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-mount-details/cephfs-mount-details.component.ts

```diff
@@ -11,25 +14,98 @@ import { MountData } from '~/app/shared/models/cephfs.model';
 export class CephfsMountDetailsComponent extends BaseModal implements OnInit {
   @ViewChild('mountDetailsTpl', { static: true })
   mountDetailsTpl: any;
+  mountDetail!: Record<string, any>;
   onCancel?: Function;
   private MOUNT_DIRECTORY = '<MOUNT_DIRECTORY>';
+  filteredUsers: CephUser[] = [];
+  selectedUser: string = 'client.admin';
+  updatedUser: string;
+  publicaddr: string;
+  secret: string;
+  userForm: CdFormGroup;
+  mount!: string;
+  cephfsmount!: string;
+  fuse!: string;
+  nfs!: string;
+  extractedKey!: string;
+  uniqueIp: string[];
+
   constructor(
-    public activeModal: NgbActiveModal,
-    @Optional() @Inject('mountData') public mountData: MountData
+    private monitorService: MonitorService,
+    public cephUserService: CephUserService,
+    @Optional() @Inject('mountData') public mountData: MountData,
   ) {
     super();
   }
-  mount!: string;
-  fuse!: string;
-  nfs!: string;
 
   ngOnInit(): void {
-    this.mount = `sudo mount -t ceph <CLIENT_USER>@${this.mountData?.clusterFSID}.${this.mountData?.fsName}=${this.mountData?.path} ${this.MOUNT_DIRECTORY}`;
-    this.fuse = `sudo ceph-fuse  ${this.MOUNT_DIRECTORY} -r ${this.mountData?.path} --client_mds_namespace=${this.mountData?.fsName}`;
-    this.nfs = `sudo mount -t nfs -o port=<PORT> <IP of active_nfs daemon>:<export_name> ${this.MOUNT_DIRECTORY}`;
+    this.mountDetail = this.mountData;
+    this.userForm = new CdFormGroup({
+      user: new UntypedFormControl(this.selectedUser, {})
+    });
+    this.getUser();
+    this.userChangeHandler();
   }
 
   cancel() {
     this.closeModal();
   }
+
+  getUser() {
+    this.cephUserService.getCephUser().subscribe((data: CephUserResponse) => {
```

Comment:
you don't need to create an extra type.
```suggestion
    this.cephUserService.getCephUser().subscribe((cephUsers: CephUser[]) => {
```